ksql-serde/src/main/java/io/confluent/ksql/serde/util/SerdeUtils.java

```diff
@@ -85,4 +88,40 @@ public static double toDouble(final Object object) {
     throw new IllegalArgumentException("This Object doesn't represent a double");
   }
 
-}
\ No newline at end of file
+  public static Schema toOptionalSchema(Schema schema) {
+    SchemaBuilder optionalBuilder = SchemaBuilder.struct();
+    for (Field field : schema.fields()) {
+      if (field.schema().type().isPrimitive()) {
+        optionalBuilder.field(
+            field.name(),
+            SchemaBuilder.type(field.schema().type()).optional().build()
+        );
+      }
+      if (field.schema().type().equals(Schema.Type.ARRAY)) {
+        optionalBuilder.field(
+            field.name(),
+            SchemaBuilder.array(
+                field.schema().valueSchema()
```

Comment:
Call `toOptionalSchema()` for the valueSchema since it can be non-primitive. Same for MAP STRUCT.